vecfixed/src/lib.rs

```diff
@@ -0,0 +1,107 @@
+//noinspection ALL
+/// VecFixed is basically a vector that keep a fixed size. Every time new element is pushed
+/// to the vector, the oldest element is removed and the latest pushed is added to the end.
+#[derive(Default)]
+pub struct VecFixed<const N: usize, T: Default + ToString> {
+    start: usize,
+    buffer: Vec<T>,
+}
+
+impl<const N: usize, T: Default + ToString> VecFixed<N, T> {
+    pub fn new() -> Self {
+        let mut buffer = Vec::with_capacity(N);
+        for _ in 0..N {
+            buffer.push(T::default());
+        }
```

Comment:
Curiosity, doesn't `let mut buffer = vec![T::default(); N]` work?

Reply:
We can also do it, but this requires `Clone` to be satisfied

Reply:
Uh ok makes sense